### ISSUE TITLE
`as` can lead to surprise mutation

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#33]
Even with both the rules `no-unsafe-assignment` and `no-unsafe-type-assertion` enabled, this isn't complained about:

```ts
type MutableA = { a: string };
type ReadonlyA = { readonly a: string };

const ro: ReadonlyA = { a: "" };
const mut = ro as MutableA; // This shouldn't be allowed
```

It's not immediately clear which of those two rules should catch this, but at least one of them should.
